### PR TITLE
Add presets management API

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
Added new endpoints:
- List presets - to list all presets
- List presets per provider
- Update status of a preset (enable/disable)
- Create a new preset
- Add/Edit provider configuration for the preset

Other changes:
- Cleaned up preset handling logic - adding a new provider will only require adding a new entry to the supported provider list.
- Added new API Preset object that provides more information (name, status, configured provider list).

**Which issue(s) this PR fixes**:
Related to #6123

**Does this PR introduce a user-facing change?**:

```release-note
Added new endpoints to list/create/update presets.
```

### CODE REVIEW

File: pkg/handler/v2/preset/preset.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+			return nil, errors.NewBadRequest(err.Error())
+		}
+
+		userInfo, err := userInfoGetter(ctx, "")
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		preset, err := presetsProvider.GetPreset(userInfo, req.Body.Name)
+		if err != nil && !k8serrors.IsNotFound(err) {
```
Comment (zreigz) [inline pkg/handler/v2/preset/preset.go]: if not found then it will be notified, not needed to return a custom error below

Comment (floreks) [reply pkg/handler/v2/preset/preset.go]: ahh... yes. I have to update the logic.

File: pkg/crd/kubermatic/v1/preset.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ func IsProviderSupported(name string) bool {
 	return false
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (floreks) [inline pkg/crd/kubermatic/v1/preset.go:71]: @zreigz my hero :heart:

**Comment by floreks** [comment]
@xrstf do you have any idea why the canary deployment check is failing?

**Comment by xrstf** [comment]
@floreks I do in fact know the problem and the solution is very simple and contained in #6122 (updating the conformance-tests Docker image to get Helm 2.17, see https://github.com/kubermatic/infra/pull/1213).

Until then

/override pre-kubermatic-canary-deployment-ci-kubermatic-io

**Comment by floreks** [comment]
@moadqassem or @xrstf PTAL as it also requires approval from crd pkg owners.

**Comment by floreks** [comment]
All green :tada: @moadqassem or @xrstf PTAL

**Comment by moadqassem** [comment]
@floreks will take a look today ;)

File: pkg/provider/kubernetes/preset.go [hunk 3]
```diff
@@ -24,16 +24,23 @@ import (
 	"os"
 	"strings"
 
-	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
-	"k8c.io/kubermatic/v2/pkg/provider"
-
+	"k8s.io/apimachinery/pkg/api/errors"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/yaml"
+
```
Comment (moadqassem) [inline pkg/provider/kubernetes/preset.go]: Please regroup the imports:
- go packages 
- github packages 
- kubermatic packages 
- kubernetes packages

File: pkg/handler/v2/preset/preset.go [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/gorilla/mux"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (moadqassem) [inline pkg/handler/v2/preset/preset.go]: Please regroup the imports:

- go packages
- github packages
- kubermatic packages
- kubernetes packages

File: pkg/handler/v2/preset/preset_test.go [hunk 5]
```diff
@@ -0,0 +1,1066 @@
...
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (moadqassem) [inline pkg/handler/v2/preset/preset_test.go]: Please regroup the imports:

- go packages
- github packages
- kubermatic packages
- kubernetes packages

**Review by moadqassem — COMMENTED** [review]
I left minor comments, related to the imports grouping

**Pull request merged**